18_links_lab/linkslab.txt

```diff
@@ -2,19 +2,32 @@ Preparation: Perform the following  (no need to write the commands you used):
 
   - create a new directory named "links"
   - cd to it
-  - inside the directory create 2 text files named "file1" and "file2". Write the names of some things you like in file "file1", and some things you don't like in file "file2"
+  - inside the directory create 2 text files named "file1" and "file2". 
+Write the names of some things you like in file "file1", and some things you don't like in file "file2"
 
+
 1. Perform the following and write the commands you used, from within links directory:
 
   - Create a symbolic link named file3 that points to "file1"
-  Answer: 
+  
+Answer: 
 
-  - Create a hard link named file4 that points to "file2"
-  Answer: 
+ ln -s file1 file3
+- Create a hard link named file4 that points to "file2"
+  
+Answer: 
 
-  - Create a new directory named "tmp" in your home folder:
-  Answer:
+  ln file2 file4
+- Create a new directory named "tmp" in your home folder:
+  
+Answer:
 
-2. Move "file3" to the new "~/tmp" directory you just created. Can you still use it to see the original "file1" ? Why not? How would you create a symbolic link that would survive the transfer to ~/tmp ?
+   mkdir /home/avia/tmp
+
+2. Move "file3" to the new "~/tmp" directory you just created. Answer: mv home/avia/lab4/links/file3 /home/avia/tmp

```

Comment:
For a symbolic link to survive the move, it has to point to an absolute path. For example the following code creates a symbolic link that ALWAYS points to /etc/shells, even if moved:

```
ln -s /etc/shells my-link 
```

So one option is to use the absolute path of the file for example:

```
ln -s /home/user/links/file1 file3 
```

But, if you're already in the directory /home/user/links (because you cd to it earlier), you can use the even shorter version:
```
ln -s $(realpath file1) file3 
```

Its meaning: Run the command realpath file1, take its output and use it as the argument. Since realpath returns the absolute path to a file it yields the same result as the longer version.